instrumentation/github.com/99designs/gqlgen/otelgqlgen/tags.go

```diff
@@ -0,0 +1,120 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otelgqlgen
+
+import (
+	"fmt"
+
+	"github.com/vektah/gqlparser/v2/ast"
+	"github.com/vektah/gqlparser/v2/gqlerror"
+
+	"go.opentelemetry.io/otel/attribute"
+)
+
+const (
+	RequestVariablesPrefix        = "request.variables"
+	ResolverArgsPrefix            = "resolver.args"
+	ResolverErrorPrefix           = "resolver.error"
+	ServiceNameKey                = attribute.Key("service.name")
+	RequestQueryKey               = attribute.Key("request.query")
+	RequestComplexityLimitKey     = attribute.Key("request.complexityLimit")
+	RequestOperationComplexityKey = attribute.Key("request.operationComplexity")
+	ResolverPathKey               = attribute.Key("resolver.path")
+	ResolverObjectKey             = attribute.Key("resolver.object")
+	ResolverFieldKey              = attribute.Key("resolver.field")
+	ResolverAliasKey              = attribute.Key("resolver.alias")
+	ResolverHasErrorKey           = attribute.Key("resolver.hasError")
+	ResolverErrorCountKey         = attribute.Key("resolver.errorCount")
+)
+
+// ServiceName defines the service name for this span.
+func ServiceName(serviceName string) attribute.KeyValue {
+	return ServiceNameKey.String(serviceName)
+}
```

Comment:
I think this should be removed. The service name should be via  "resource"https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/README.md by setting `OTEL_RESOURCE_ATTRIBUTES` or `OTEL_SERVICE_NAME` env var. It is possible to do it via code as well see:  [example](https://github.com/open-telemetry/opentelemetry-go/blob/main/example/fib/main.go)